runtime/array.c

```diff
@@ -346,6 +335,25 @@ CAMLprim value caml_make_vect(value len, value init)
   CAMLreturn (res);
 }
 
+#ifndef FLAT_FLOAT_ARRAY
+static value uninitialized_float = Val_unit;
+#endif
+
+/* [len] is a [value] representing number of floats */
+/* [ int -> float array ] */
+CAMLprim value caml_make_float_vect(value len)
+{
+#ifdef FLAT_FLOAT_ARRAY
+  return caml_floatarray_create (len);
+#else
+  if (uninitialized_float == Val_unit){
+    caml_register_global_root (&uninitialized_float);
```

Comment:
Could this be a generational root to avoid the tiny (few ns) hit of scanning it on every minor collection?

Reply:
Done.